internal/controllers/machinedeployment/machinedeployment_rollout_sequence_test.go

```diff
@@ -0,0 +1,876 @@
+		for _, taskID := range taskOrder {
+			if taskID == 0 {
+				fileLogger.Logf("[MD controller] Iteration %d, Reconcile md", i)
+				fileLogger.Logf("[MD controller] - Input to rollout planner\n%s", current)
+
+				// Running a small subset of MD reconcile (the rollout logic and a bit of setReplicas)
+				p := newRolloutPlanner()
+				p.md = current.machineDeployment
+				p.newMS = current.newMS()
+				p.oldMSs = current.oldMSs()
+
+				err := p.Plan(ctx)
+				g.Expect(err).ToNot(HaveOccurred())
+				// Apply changes.
+				for _, ms := range current.machineSets {
+					if scaleIntent, ok := p.scaleIntents[ms.Name]; ok {
+						ms.Spec.Replicas = ptr.To(scaleIntent)
+					}
+				}
+				current.machineDeployment.Status.Replicas = mdutil.GetActualReplicaCountForMachineSets(current.machineSets)
+				current.machineDeployment.Status.AvailableReplicas = mdutil.GetAvailableReplicaCountForMachineSets(current.machineSets)
+
+				// Log state after this reconcile
+				fileLogger.Logf("[MD controller] - Result of rollout planner\n%s", current)
+
+				// Check we are not breaching rollout constraints
+				minAvailableReplicas := ptr.Deref(current.machineDeployment.Spec.Replicas, 0) - mdutil.MaxUnavailable(*current.machineDeployment)
+				totAvailableReplicas := ptr.Deref(current.machineDeployment.Status.AvailableReplicas, 0)
+				if totAvailableReplicas < minAvailableReplicas {
+					tolerateBreach := false
+					if tt.minAvailableBreachToleration != nil {
+						tolerateBreach = tt.minAvailableBreachToleration(fileLogger, i, current, minAvailableReplicas, totAvailableReplicas)
+					}
+					if !tolerateBreach {
+						g.Expect(totAvailableReplicas).To(BeNumerically(">=", minAvailableReplicas), "totAvailable machines is less than MinUnavailable")
+					}
+				}
+
+				maxAllowedReplicas := ptr.Deref(current.machineDeployment.Spec.Replicas, 0) + mdutil.MaxSurge(*current.machineDeployment)
+				totReplicas := mdutil.TotalMachineSetsReplicaSum(current.machineSets)
+				if totReplicas > maxAllowedReplicas {
+					tolerateBreach := false
+					if tt.maxSurgeBreachToleration != nil {
+						tolerateBreach = tt.maxSurgeBreachToleration(fileLogger, i, current, maxAllowedReplicas, totReplicas)
+					}
+					if !tolerateBreach {
+						g.Expect(totReplicas).To(BeNumerically("<=", maxAllowedReplicas), "totReplicas machines is greater than MaxSurge")
+					}
+				}
+			}
+
+			// Run mutators faking other controllers
+			for _, ms := range current.machineSets {
+				if fmt.Sprintf("ms%d", taskID) == ms.Name {
+					fileLogger.Logf("[MS controller] Iteration %d, Reconcile ms%d, %s", i, taskID, msLog(ms, current.machineSetMachines[ms.Name]))
+					machineSetControllerMutator(fileLogger, ms, current)
+					break
+				}
+			}
+		}
+
+		// Check if we are at the desired state
+		if current.Equal(desired) {
+			fileLogger.Logf("[Test] Final state\n%s", current)
+			break
+		}
+
+		// Safeguard for infinite reconcile
+		i++
+		if i > maxIterations {
+			current.Equal(desired)
```

Comment:
```suggestion
```
Not needed, this didn't change since l.385 (also the result of this is not handled anyway)